tests/core/pyspec/eth2spec/test/whisk/block_processing/test_process_shuffled_trackers.py

```diff
@@ -0,0 +1,130 @@
+from eth2spec.test.context import spec_state_test, with_whisk_and_later, expect_assertion_error
+from eth2spec.test.helpers.keys import whisk_ks_initial
+from eth2spec.test.helpers.whisk import compute_whisk_tracker
+from curdleproofs import GenerateWhiskShuffleProof
+
+
+def set_correct_shuffle_proofs(spec, state, body):
+    pre_shuffle_trackers = get_and_populate_pre_shuffle_trackers(spec, state, body)
+
+    post_trackers, shuffle_proof = GenerateWhiskShuffleProof(spec.CURDLEPROOFS_CRS, pre_shuffle_trackers)
+    body.whisk_post_shuffle_trackers = post_trackers
+    body.whisk_shuffle_proof = shuffle_proof
+
+
+def get_and_populate_pre_shuffle_trackers(spec, state, body):
+    shuffle_indices = spec.get_shuffle_indices(body.randao_reveal)
+    pre_shuffle_trackers = []
+    for i in shuffle_indices:
+        # Set r to some value > 1 ( = 2+i)
+        tracker = compute_whisk_tracker(whisk_ks_initial[i], 2 + i)
+        state.whisk_candidate_trackers[i] = tracker
+        pre_shuffle_trackers.append(tracker)
+    return pre_shuffle_trackers
+
+
+def get_pre_shuffle_trackers(spec, state, body):
+    return [state.whisk_candidate_trackers[i] for i in spec.get_shuffle_indices(body.randao_reveal)]
+
+
+def set_state_epoch(spec, state, epoch):
+    state.slot = epoch * spec.SLOTS_PER_EPOCH
+
+
+def set_state_epoch_selection_gap(spec, state):
+    set_state_epoch(spec, state, spec.WHISK_EPOCHS_PER_SHUFFLING_PHASE - 1)
```

Comment:
`WHISK_EPOCHS_PER_SHUFFLING_PHASE` is in config, so here we should add `spec.config` in testing:
```suggestion
    set_state_epoch(spec, state, spec.config.WHISK_EPOCHS_PER_SHUFFLING_PHASE - 1)
```